### PR TITLE
Send additional RRs when PTR and SRV are requested

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
We don't send additional records when PTR or SRV records are requested.  This is listed as a SHOULD in RFC 6763, so we might as well do so.

#### Change overview
Sends specified additional records when PTR or SRV records are requested.

#### Testing
avahi-browse to request PTR, checked packed contents with wireshark.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: src/lib/dnssd/minimal_mdns/QueryReplyFilter.h [hunk 1]
```diff
@@ -70,8 +70,22 @@ class QueryReplyFilter : public ReplyFilter
         {
             return true;
         }
-
-        return ((mQueryData.GetType() == QType::ANY) || (mQueryData.GetType() == qType));
+        if ((mQueryData.GetType() == QType::ANY) || (mQueryData.GetType() == qType))
```
Comment (andy31415) [inline src/lib/dnssd/minimal_mdns/QueryReplyFilter.h:73]: How about #12413 - that  one is doing the same thing  and allowing anything for additional (and the rules on what to send additional are encoded in the responder setup).

Is this change still needed?
